### PR TITLE
Check for cycles

### PR DESCRIPTION
**Pull request by chris-armstrong**
* Upgrade jasmine version
* Cycle check on load() and re-order so that addResource / addRole will work with parent references further down the list.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -42,6 +42,67 @@ function set(obj, path, value) {
   obj[components[components.length - 1]] = value;
 }
 
+function find(list, predicate) {
+  // 1. Let O be ? ToObject(list value).
+  if (list == null) {
```
Comment (northern) [inline index.js]: Maybe strict equality?

File: index.js [hunk 2]
```diff
@@ -42,6 +42,67 @@ function set(obj, path, value) {
...
+    throw new TypeError('"list" is null or not defined');
```
Comment (northern) [inline index.js:48]: Not sure if `TypeError` is the correct one to throw here but I guess it doesn't matter... :)

File: index.js [hunk 3]
```diff
@@ -42,6 +42,67 @@ function set(obj, path, value) {
...
+
+  // 2. Let len be ? ToLength(? Get(O, "length")).
+  var len = o.length >>> 0;
+
+  // 3. If IsCallable(predicate) is false, throw a TypeError exception.
+  if (typeof predicate !== 'function') {
+    throw new TypeError('predicate must be a function');
+  }
+
+  // 4. If thisArg was supplied, let T be thisArg; else let T be undefined.
```
Comment (northern) [inline index.js:61]: Probably meant to be "If arguments was supplied, "..

**Review by northern — APPROVED** [review]
Since the original tests haven't been modified, I'd give this 👍 (as if it matters, lol). Cool you guys are still using this...

File: index.js (hunk 3 above)
Comment (chris-armstrong) [reply index.js:61]: The code is lifted from MDN's definition for Array.find() - I'll clean it up a little, but in the future will need to do the ES6 upgrade to the code as per #4

**Review by em0ney — APPROVED** [review]
👍

**Pull request merged**